src/context.rs

```diff
@@ -263,16 +266,40 @@ impl Context {
         } else {
             return Err(Error::WrapperError(ErrorKind::WrongValueFromTpm));
         };
-        let capabilitydata = unsafe { MBox::from_raw(outcapabilitydata) };
         let ret = Error::from_tss_rc(ret);
 
         if ret.is_success() {
-            Ok((*capabilitydata, moredata))
+            let capabilitydata = unsafe { MBox::from_raw(outcapabilitydata) };
+            let capabilities = CapabilityData::try_from(*capabilitydata)?;
+            Ok((capabilities, moredata))
         } else {
             Err(ret)
         }
     }
 
+    /// Determine a TPM property, which is automatically cached if available
+    pub fn get_tpm_property(&mut self, property: u32) -> Result<Option<u32>> {
```

Comment:
What is that ```property```?

If that is property tag then I would rather have a type or an enum for it like ```StructureTag``` (src/constants/tags/structure.rs) unless it would be an excessive amount of work? If it is to much work then at least do like make a ```PropertyTag ``` type that only wraps a u32 value and provides conversions to and from TPM_PT. And then make a ticket to make a proper implementation of ```PropertyTag```.

The reason I would like it is because this is a public api and we should try to make it hard to use them incorrectly

Reply:
Will do.